### ISSUE TITLE
LayoutReader：the gt should be word level or paragraph level

### ISSUE THREAD

**Issue by myh12138** [microsoft/unilm#566]
**Describe**
Model I am using (Layoutreader.):
thanks for your great contribution in reading order. During reading the paper and code i have some questions,
the paper show  paragraph level effects and sentence level effects. But the ReadingBank datasets show word level mark，can you make a more clear for these,thanks.

**Comment by myh12138** [comment]
@HYPJUDY  @zlwang-cs

**Comment by zlwang-cs** [comment]
Hi, thanks for your comments. I am not sure about the paragraph level and sentence level effects you mentioned. In fact, our works (both LayoutReader & ReadingBank) are about work level which is the most fine-grained and reliable (since paragraph segmentation may be problematic). We also show how to adapt our word-level results to textline-level (see Section 5.6) but the first thing is still to extract the word-level order.

**Comment by myh12138** [comment]
> Hi, thanks for your comments. I am not sure about the paragraph level and sentence level effects you mentioned. In fact, our works (both LayoutReader & ReadingBank) are about work level which is the most fine-grained and reliable (since paragraph segmentation may be problematic). We also show how to adapt our word-level results to textline-level (see Section 5.6) but the first thing is still to extract the word-level order.

Thanks for your reply, I understand your mean,you do a lot of post-processing to get the effect figure 1 and figure 4 in your paper to make the order form word-level to  paragraph -level. I will try it latter thanks.

**Comment by vogaliccb** [comment]
> 
So I must use an OCR engines which can do word-level recognization

**Comment by zlwang-cs** [comment]
> So I must use an OCR engines which can do word-level recognization

Yes, our model is focusing on the word-level reading order so the word-level information is necessary.

**Comment by warren-wzw** [comment]
Hello author, first of all, thank you for your work. In addition, I would like to ask if the level of code is open source，I look forward to your reply

**Comment by rajat-008** [comment]
> Hi, thanks for your comments. I am not sure about the paragraph level and sentence level effects you mentioned. In fact, our works (both LayoutReader & ReadingBank) are about work level which is the most fine-grained and reliable (since paragraph segmentation may be problematic). We also show how to adapt our word-level results to textline-level (see Section 5.6) but the first thing is still to extract the word-level order.

@zlwang-cs 
In figure 1 of your paper you represent the paragraph level reading order. I would like to understand if layoutreader is capable of identifying paragraph level reading order, if the input is the bounding boxes of these paragraphs?

Thanks in advance!

**Comment by zlwang-cs** [comment]
@rajat-008
Yes, LayoutReader is capable of solving paragraph-level reading order. You can check Section 5.6 where we describe how to sort the paragraphs by the word-level order.

**Comment by rajat-008** [comment]
> @rajat-008 Yes, LayoutReader is capable of solving paragraph-level reading order. You can check Section 5.6 where we describe how to sort the paragraphs by the word-level order.

Hi @zlwang-cs 

Thank you for the reply. I did go through the section 5.6 and I understand that layoutreader is capable of solving reading order if the test-line bounding boxes are given. 

Is it possible to achieve the same without passing the text information and just pass the bounding boxes of the paragraphs? To put it another way - if the given input is just the bounding boxes of paragraphs(the text contained in these paragraphs is not input), is layoutreader capable of sorting the reading order of these paragraphs?

Thanks in advance!

**Comment by zlwang-cs** [comment]
@rajat-008 I understand your point. Unfortunately, it is hard to predict the paragraph order through the bbox of paragraphs for LayoutReader. As you can see from the paper, LayoutReader is a "word-level" model, which means the expected input and output are words/tokens. We can sort the paragraphs by the more fine-grained word order. 

One thing you can try is to replace the word bbox with the paragraph bbox. I believe it is possible to run the model on the newer setting. Then each paragraph is treated as a very wide word. However, I highly doubt the performance. Overall, I recommend you trying new fine-tuning strategy to better adapt the model into your application scenario.
